### ISSUE TITLE
[proposal] Add the koordinator-root-quota CRD in koordinator and include the koordinator-root-quota information in quota debug API.

### ISSUE THREAD

**Issue by tan90github** [koordinator-sh/koordinator#1519]
**What is your proposal**:

- Background
  - When I use [quota debug API](https://koordinator.sh/docs/user-manuals/multi-hierarchy-elastic-quota-management/#quota-debug-api) in the ElasticQuota plugin to view quotas information, I can query koordinator-default-quota and koordinator-system-quota, but there is no information about koordinator-root-quota.
![image](https://github.com/koordinator-sh/koordinator/assets/81003453/da1948b9-e0a2-4ea6-9f2b-57e2cdeb1c98)

- Proposal
  - We hope to add the koordinator-root-quota CRD in koordinator and include the koordinator-root-quota information in quota debug API.(apis/v1/plugins/ElasticQuota/quotas).

**Why is this needed**:

- Adding this information to the debug API of the ElasticQuota plugin can make it more convenient for users to observe the quota allocation information, mainly for runtime information. Moreover, from the perspective of logical completeness, the quotas information in the debug API should also include information related to koordinator-root-quota.

**Is there a suggested solution, if so, please add it**:

- Solution: Similar to koordinator-default-quota and koordinator-system-quota, once the groupQuotaManager is created, create the koord-root-quota CRD in the code. In this way, the corresponding quota information can be directly viewed in the debug API.

**Comment by eahydra** [comment]
koordinator-root-quota is special, it is approximately equal to the total capacity of a cluster. If you create an ElasticQuota CRD object representation in the cluster, but it should be read-only externally, you should consider disallowing users to modify the object. If you just want to view it through the debug api, you don't need to create a CRD object. In summary, it seems that having a real CRD object representing the root quota makes the design of the current Quota mechanism more uniform.

WDYT @hormes @FillZpp @buptcozy @xulinfei1996

**Comment by hormes** [comment]
it's ok to create a root CRD. A related question is, some time, if the root quota only wants to manage part of the nodes, do we have any solutions to support it?

**Comment by eahydra** [comment]
> 

For example, nodes in a cluster need to be divided into multiple categories, and each category may have a Root Quota, which leads to subsequent changes in the model of the entire Quota Tree; but if there is and only one Root Quota, it is simpler, at least we can constrain node ranges on ElasticQuota Annotation.

**Comment by eahydra** [comment]
> it's ok to create a root CR. A related question is, some time, if the root quota only wants to manage part of the nodes, do we have any solutions to support it?

For example, nodes in a cluster need to be divided into multiple categories, and each category may have a Root Quota, which leads to subsequent changes in the model of the entire Quota Tree; but if there is and only one Root Quota, it is simpler, at least we can constrain node ranges on ElasticQuota Annotation.

**Comment by eahydra** [comment]
@tan90github Hi, what do you think of the discussion above. If there is no problem, it may be possible to implement it according to your plan and discuss it.

**Comment by tan90github** [comment]
> @tan90github Hi, what do you think of the discussion above. If there is no problem, it may be possible to implement it according to your plan and discuss it.

Hi, thanks for your reply. Based on the above discussion, I have the following opinions:
1. I think that if the root-quota only wants to manage part of the nodes, the cluster administrator should create quota resources for different categories as its "root-quota".  At this time, the structure of the entire "Quota Tree" is as shown in the following figure. For different categories of nodes, the administrator should adjust their corresponding category-root-quota. This can solve the problem discussed above.
![image](https://github.com/koordinator-sh/koordinator/assets/81003453/0f324d16-cdcb-4df2-a7ba-0c3af7875428)

2. If there are multiple root-quota related requirements in the future, I think a separate issue can be raised and discussed.

In summary, I currently prefer to implement based on having only one root-quota.

**Comment by tan90github** [comment]
@eahydra Hi, thank you very much for your attention to this issue. Do you have any other questions about this solution?
